src/linear_algebra/matrix.lean

```diff
@@ -203,22 +204,33 @@ def diag : (matrix n n M) →ₗ[R] n → M := {
   (diag : (matrix n n R) →ₗ[R] n → R) 1 = λ i, 1 := by {
     dunfold diag, ext, simp [one_val_eq], }
 
-protected lemma smul_sum {α : Type u} {s : finset α} {f : α → M} {c : R} :
-  c • s.sum f = s.sum (λx, c • f x) := (s.sum_hom _).symm
```

Comment:
Why isn't `smul_sum` needed any more?

Reply:
Because of https://github.com/leanprover-community/mathlib/pull/1910 which was merged almost immediately before (though review in parallel) to the change where I introduced `smul_sum`.

(I added a remark about this the second commit on this branch.)